src/thing/get.test.ts

```diff
@@ -1139,7 +1141,97 @@ describe("getStringWithLocale", () => {
   });
 });
 
-describe("getStringsWithLocaleAll", () => {
+describe("getStringByLocale", () => {
+  function makeQuad(
+    subject: string,
+    predicate: string,
+    value: string,
+    locale: string | NamedNode | undefined
+  ): Quad {
+    return DataFactory.quad(
+      DataFactory.namedNode(subject),
+      DataFactory.namedNode(predicate),
+      DataFactory.literal(value, locale)
+    );
+  }
+
+  it("returns all the string values for the given Predicate keyed by their locale", () => {
+    const subject = "https://arbitrary.vocab/subject";
+    const predicate = "https://some.vocab/predicate";
+    const thing = dataset();
+
+    thing
+      .add(makeQuad(subject, predicate, "value 1", "en"))
+      .add(makeQuad(subject, predicate, "value 2", "en"))
+      .add(makeQuad(subject, predicate, "value 3", "fr"))
+      .add(makeQuad(subject, predicate, "value 4", "es"));
+
+    const thingWithLocaleStrings = Object.assign(thing, {
+      internal_url: "https://arbitrary.vocab/subject",
+    });
+
+    expect(
+      Array.from(getStringByLocaleAll(thingWithLocaleStrings, predicate))
+    ).toEqual([
+      ["en", ["value 1", "value 2"]],
+      ["fr", ["value 3"]],
+      ["es", ["value 4"]],
+    ]);
+  });
+
+  it("includes non-language string values keyed with the empty language tag", () => {
```

Comment:
I don't have a problem with this, but for consistency we should make sure that `getStringWithLocale` returns `xsd:string` when the parameter is an empty string.

Reply:
Actually yeah, I've changed this now to remove this special handling for empty locale values - it just keeps everything simpler and more explicit.
In fact, perhaps this would be a reason to expose a `getStringAll()` function that actually returned all string values, keys on locale, and in this case also include the `xsd:string` values keyed on either an empty locale, or maybe some proprietary constant value (like "NoLocaleProvided", or whatever).  I don't like the idea of the key here extending beyond the simple RFC notion of a language tag though, so would need to discuss - but lets see if anyone ever asks for such a function first :) !